test/InputOutput/unit_spectralelement2d.jl

```diff
@@ -67,31 +63,86 @@ end
     Nq = 4
     quad = Quadratures.GLL{Nq}()
     mesh = Meshes.RectilinearMesh(domain, n1, n2)
+    grid_topology = Topologies.Topology2D(context, mesh)
+
+    return Spaces.SpectralElementSpace2D(
+        grid_topology,
+        quad;
+        enable_bubble,
+        enable_mask,
+    )
+end
+
+function mktempfile(f)
+    mktempdir() do dir
+        cd(dir) do
+            f(tempname(dir))
+        end
+    end
+end
+
+@testset "restart test for 2D spectral element simulations" begin
     device = ClimaComms.device()
     @info "Using device" device
-    context = ClimaComms.SingletonCommsContext(device)
-    grid_topology = Topologies.Topology2D(context, mesh)
+    context = ClimaComms.context(device)
+    parameters = (
+        ϵ = 0.1,  # perturbation size for initial condition
+        l = 0.5, # Gaussian width
+        k = 0.5, # Sinusoidal wavenumber
+        ρ₀ = 1.0, # reference density
+        c = 2,
+        g = 10,
+        D₄ = 1e-4, # hyperdiffusion coefficient
+    )
+    space = init_space(context; enable_bubble = true, enable_mask = false)
+    y0 = init_state.(Fields.local_geometry_field(space), Ref(parameters))
+    Y = Fields.FieldVector(y0 = y0)
 
-    for enable_bubble in (true, false)
-        space =
-            Spaces.SpectralElementSpace2D(grid_topology, quad; enable_bubble)
+    # write field vector to hdf5 file
+    mktempfile() do filename
+        InputOutput.HDF5Writer(filename, context) do writer
+            InputOutput.write!(writer, "Y" => Y) # write field vector from hdf5 file
+        end
+        Cache.clean_cache!()
+        InputOutput.HDF5Reader(filename, context) do reader
+            restart_Y = InputOutput.read_field(reader, "Y") # read fieldvector from hdf5 file
+            @test restart_Y == Y # test if restart is exact
+            @test axes(restart_Y) == axes(Y) # test if restart is exact for space
+        end
+    end
 
-        y0 = init_state.(Fields.local_geometry_field(space), Ref(parameters))
-        Y = Fields.FieldVector(y0 = y0)
+    # Test with masks
+    space = init_space(context; enable_bubble = true, enable_mask = true)
+    y0 = init_state.(Fields.local_geometry_field(space), Ref(parameters))
+    Y = Fields.FieldVector(y0 = y0)
 
-        # write field vector to hdf5 file
-        filename = tempname(pwd())
+    Spaces.set_mask!(space) do coords
+        rand() > 0.5
+    end
+    is_active₀ = Spaces.get_mask(space).is_active
```

Comment:
This doesn't seem to be used anywhere

Reply:
Ah, right, I think this is sort of a rebase artifact